server/specification/specification.go

```diff
@@ -0,0 +1,225 @@
+package specification
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"sync"
+	"text/template"
+
+	"github.com/SKF/go-rest-utility/problems"
+	"go.opencensus.io/trace"
+)
+
+const (
+	swaggerUIDirectory = "swagger-ui"
+	indexFilePath      = "swagger-ui/index.html"
+	openAPIPath        = "/openapi.yaml"
+)
+
+type SwaggerFS struct {
+	FS fs.FS
+
+	SwaggerURLPrefix      string
+	APIEndpointsURLPrefix string
+
+	NotFoundHandler         http.Handler
+	MethodNotAllowedHandler http.Handler
+
+	initOnce      sync.Once
+	compiledFiles map[string]CompiledFile
+}
+
+type swaggerFSHandler struct {
+	*SwaggerFS
+	*Config
+	pathsToCompile []string
+}
+
+type CompiledFile struct {
+	Contents *bytes.Buffer
+	Info     fs.FileInfo
+}
+
+func (swaggerfs *SwaggerFS) Handler(opts ...Option) *swaggerFSHandler {
+
+	config := &Config{
```

Comment:
Could these be public fields on the SwaggerFS struct instead? and set to defaults if they are empty?

Reply:
I think like when a developer doesn't have to care about defaults.
If just looking at the struct the developer might think they have to set those fields(or read the documentation)

Reply:
I think comments on the fields are good enough. I need to understand what they mean anyway? The usual case is that they copy the implementation from some example or another service, and if they want more information they will go into that module and investigate some.

But this works if you feel strongly about it.